### PR TITLE
fix: [PIE-3952]: expression wrong value

### PR DESCRIPTION
**Pull request by dhruvUpadhyay07**
##### Jira Links:

https://user-images.githubusercontent.com/64455881/172127897-b88801d6-079a-405d-b91a-92f5674dfa26.mov

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/HttpStep/HttpStep.tsx [hunk 1]
```diff
@@ -200,6 +200,8 @@ export class HttpStep extends PipelineStep<HttpStepData> {
         method:
           getMultiTypeFromValue(initialValues.spec?.method as string) === MultiTypeInputType.RUNTIME
             ? (initialValues.spec?.method as string)
+            : getMultiTypeFromValue(initialValues.spec?.method as string) === MultiTypeInputType.EXPRESSION
```
Comment (vkbansal) [inline src/modules/75-cd/components/PipelineSteps/HttpStep/HttpStep.tsx:203]: Can we store it in a variable and reuse? Both `initialValues.spec?.method` and  `getMultiTypeFromValue(initialValues.spec?.method as string)`

Comment (vkbansal) [reply src/modules/75-cd/components/PipelineSteps/HttpStep/HttpStep.tsx:203]: It will also reduce branches

**Comment by dhruvUpadhyay07** [comment]
retrigger titlecheck

File: src/modules/75-cd/components/PipelineSteps/HttpStep/HttpStep.tsx [hunk 2]
```diff
@@ -193,17 +193,21 @@ export class HttpStep extends PipelineStep<HttpStepData> {
...
+          type === MultiTypeInputType.RUNTIME
+            ? (methodValue as string)
+            : type === MultiTypeInputType.EXPRESSION
+            ? methodValue
             : // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-              httpStepType.find(step => step.value === (initialValues.spec?.method || 'GET'))!,
+              httpStepType.find(step => step.value === (methodValue || 'GET'))!,
         headers:
-          getMultiTypeFromValue(initialValues.spec?.headers as string) === MultiTypeInputType.RUNTIME
+          type === MultiTypeInputType.RUNTIME
```
Comment (vkbansal) [inline src/modules/75-cd/components/PipelineSteps/HttpStep/HttpStep.tsx:210]: this is headers

File: src/modules/75-cd/components/PipelineSteps/HttpStep/HttpStep.tsx [hunk 3]
```diff
@@ -216,7 +220,7 @@ export class HttpStep extends PipelineStep<HttpStepData> {
             ? undefined
             : [],
         outputVariables:
-          getMultiTypeFromValue(initialValues.spec?.outputVariables as string) === MultiTypeInputType.RUNTIME
+          type === MultiTypeInputType.RUNTIME
```
Comment (vkbansal) [inline src/modules/75-cd/components/PipelineSteps/HttpStep/HttpStep.tsx:223]: variables

**Review by vkbansal — CHANGES_REQUESTED** [review]

**Review by vkbansal — APPROVED** [review]

**Review by samarth-sharma31 — APPROVED** [review]

**Pull request merged**